benchmark-sets/comparison/abseil-cpp.yaml

```
@@ -0,0 +1,16 @@
+"functions":
+- "name": "absl::str_format_internal::FormatArgImpl::Dispatch<long long>(absl::str_format_internal::FormatArgImpl::Data, absl::str_format_internal::FormatConversionSpecImpl, void*)"
```

Should params be removed from name?
```suggestion
- "name": "absl::str_format_internal::FormatArgImpl::Dispatch<long long>"
```

---

Not for now if we want to omit the return type in names.
Please correct me if I am wrong: In C++, two functions can have the same name (i.e., `Dispatch `) and template specialization (i.e., `<long long>`) but different return types and params.
Happy to omit the params if I am wrong, otherwise we have to have either the return types or params.

BTW, may I ask why you would prefer to omit the parameters?

---

Assuming we don't intend to use function name as the unique identifier, we already have separated params info below, and params seems unrelated to the name of the function. Function signature can be used to identify a specific overloaded function, or by reconstructing `"params"`.
Since function params are removed in some projects (e.g. [avahi](https://github.com/google/oss-fuzz-gen/pull/64/files#diff-80970d1beea041fc1ab131007204780451bf90ede303cc4fc2c2dad89e7c8568)), it'd be nice to keep with a same format (with or without params)?

---

1. Yes, you are right: we will use signature as the unique identifier in APIs later.
2. However, the function name also needs to be unique for now because they will be used as [workdir names](https://github.com/google/oss-fuzz-gen/blob/1a0c8323b3b00701ca0910ea749fecc8857fb791/run_all_experiments.py#L81).
3. Later, we can resolve this by separating three things:
  a) benchmark id, 
  b) benchmark directory name (which needs to be unique), and 
  c) benchmark docker image (which needs to be unique and should not contain special chars). 
  But I reckon this is not related to the primary intention of this PR (making the benchmark more structured) or our priority. So far, we only have ugly names in the report, and need a bit parsing when checking if function is in fuzz target (Please do let me know if [this regex](https://github.com/google/oss-fuzz-gen/pull/64/commits/d813d36d38d479a9e291041f1f34de623010bcec#diff-2988c9b2abbf23c03fa8ed236ed0f62fb249e13ef31c1509b7fef0f6d78dd1beR127) does not work).
4. Yes, params are removed from some projects. That's mainly because they are C programs, which do not allow overloading.

---

I got some time to extend this PR a bit more based on your suggestions:
1. Beautify benchmark names. Now benchmark directory name, which is still `becnhmark.id` uses `<return_type><function_name>< Template Specialization>`. This removes params, making the directory shorter and more readable while maintaining uniqueness.
2. Separate the docker image name from `benchmark.id` by [removing the special chars](https://github.com/google/oss-fuzz-gen/pull/64/commits/770cafdf133df5b7adc6c3847326341d345ea097) in them.
3. Ensure special chars are encoded in the HTML report so that we can download them via `wget2` and visit them.

Here is the [final view](https://github.com/google/oss-fuzz-gen/pull/64/commits/770cafdf133df5b7adc6c3847326341d345ea097).

---

I might use the function signature as benchmark id later for simplicity, as we will use that as the unique identifier in FI anyway.